pkg/cvo/upgradeable.go

```diff
@@ -263,6 +274,45 @@ func (check *clusterVersionOverridesUpgradeable) Check() *configv1.ClusterOperat
 	return cond
 }
 
+type upgradeInProgressUpgradeable struct {
```

Comment:
@wking 
I am thinking if it is a good choice to use un `Upgradeable` type here?
Non-nil return from the `Check()` function for all the existing `Upgradeable`s [leads to](https://github.com/openshift/cluster-version-operator/blob/e546515213c8681ca44c52f178401cd47ad07d11/pkg/cvo/upgradeable.go#L97) `Upgradeable=False`

But this new one does not. It depends on the 2nd upgrade request from the user and we only block it if it is minor level.
I feel pushing all the logic here to precondition is more reasonable.